app/Http/Controllers/Service/Admin/TradersController.php

```diff
@@ -224,12 +225,17 @@ public function update(AdminNewUpdateTraderRequest $request, int $id)
                     // to keep SQLite happy
                     ->groupBy('id')
                     ->having('traders_count', '=', 0)
-                    ->pluck('id')
-                    ->toArray();
-
-                // ... and soft-delete them, since they are orphans
-                User::whereIn('id', $orphanUsers)->delete();
-
+                    ->get();
+
+                // this is a slow iteration, but we don't expect too many users/trader
+                $orphanUsers->each(function ($orphanUser) {
+                    // make a Uuid for the email, which has a unique constraint
+                    $hash = Uuid::uuid4()->toString();
+                    // localhost - it gets emailed, somehow, it'll be looped back to the box.
+                    $orphanUser->update(['email' => $hash . '@localhost']);
```

Comment:
just curious about the `@localhost` bit 🌵

Reply:
I'm inventing a "valid", unique but undeliverable email address.

An email which is whatever@localhost *should* be routed to the current box. 
localhost is a reserved domain, and the internet won't deliver it.

It's arguable that we could just have stuck anything in the email field, but i'd prefer if it was at lease emaily.